apps/web/middleware.ts

```diff
@@ -1,8 +1,28 @@
+import { getToken } from "next-auth/jwt";
 import { collectEvents } from "next-collect/server";
+// eslint-disable-next-line @next/next/no-server-import-in-page
+import { NextMiddleware, NextResponse } from "next/server";
 
 import { extendEventData, nextCollectBasicSettings } from "@calcom/lib/telemetry";
 
+const middleware: NextMiddleware = async (req) => {
+  const url = req.nextUrl;
+  const token = await getToken({ req });
+
+  /**
+   *  TODO: Add logic to only redirect certain pages if `v2` cookie is present.
+   * As of now, only admins can view the new v2 pages
+   **/
+  if (token?.role === "ADMIN" && url.pathname.startsWith(`/settings/admin`)) {
+    // rewrite to the current subdomain under the pages/sites folder
+    url.pathname = `/v2${url.pathname}`;
+  }
```

Comment:
Only admins and admin routes will be served from v2 for now

Reply:
This would be super easy to detect if the user has a cookie present right?

Reply:
yup @sean-brydon that's my next step